src/coordination/coordinator_state_machine.cpp

```diff
@@ -12,14 +12,136 @@
 #ifdef MG_ENTERPRISE
 
 #include "nuraft/coordinator_state_machine.hpp"
+#include "utils.hpp"
+#include "utils/logging.hpp"
+
+#include <regex>
 
 namespace {
+constexpr std::string_view kCoordClusterState = "coord_cluster_state";
+constexpr std::string_view kLastLogIdx = "last_log_idx";
+constexpr std::string_view kLastLogTerm = "last_log_term";
+constexpr std::string_view kSize = "size";
+constexpr std::string_view kLastConfig = "last_config";
+constexpr std::string_view kType = "type";
+constexpr std::string_view kSnapshotIdPrefix = "snapshot_id_";                // top level
+constexpr std::string_view kSnapshotVersion = "snapshot_durability_version";  // top level
+constexpr int kActiveVersion = 1;
 constexpr int MAX_SNAPSHOTS = 3;
+
+nuraft::snapshot::type GetSnapshotType(int val_type) {
+  // NOLINTNEXTLINE
+  switch (val_type) {
+    case 1:
+      return nuraft::snapshot::raw_binary;
+    case 2:
+      return nuraft::snapshot::logical_object;
+  }
+  LOG_FATAL("Val type unknown");
+}
+
+int FromSnapshotType(nuraft::snapshot::type val_type) {
+  switch (val_type) {
+    case nuraft::snapshot::raw_binary:
+      return 1;
+    case nuraft::snapshot::logical_object:
+      return 2;
+  }
+  LOG_FATAL("Unknown snapshot type");
+}
```

Comment:
1. ) constexpr it
2. I don't think you even need this functions, you can just static_cast